sopn_publish_date/__init__.py

```diff
@@ -19,23 +23,50 @@ def __init__(self):
             "gla": self.greater_london_assembly,
             "pcc": self.police_and_crime_commissioner,
             "mayor": self.mayor,
-            "parl": self.uk_parliament,
         }
         self.calendar = UnitedKingdomBankHolidays()
 
-    def for_id(self, election_id: str) -> date:
+    def for_id(self, election_id: str, country: str = None) -> date:
         """
-        Calculate the publish date for an election given in `uk-election-ids <https://elections.democracyclub.org.uk/reference_definition/>`_ format, or raise an exception if that election id is ambiguous (could correspond to elections in multiple countries with different electoral legislation)
+        Calculate the publish date for an election given in `uk-election-ids <https://elections.democracyclub.org.uk/reference_definition/>`_ format and an optional country if necessary (for example, local or parliamentary elections), or raise an exception if that election id is ambiguous (could correspond to elections in multiple countries with different electoral legislation)
 
         :param election_id: a string representing an election id in uk-election-ids format
+        :param country: an optional string representing the country where the election will be held
         :return: a datetime representing the expected publish date
         """
+
         election_type, poll_date = type_and_poll_date(election_id)
 
+        countries = {
+            "ENG": Country.ENGLAND,
+            "WLS": Country.WALES,
+            "SCT": Country.SCOTLAND,
+            "NIR": Country.NORTHERN_IRELAND,
```

Comment:
I can why you're doing this here, but if we take a step back and think about the public interface of this library, its a bit strange that `.local()` and `.uk_parliament()` take a `calendars.Country` to describe the country, but `.for_id()` takes a string instead for.. reasons?

I reckon its reasonable to either say its up to the client application to pass in the right constant or allow either format everywhere, as opposed to just this one function. Either is fine really, but it would be nice to be consistent.

Reply:
Good point, folded into the latest commit - the method take a `Country` instead and the onus of mapping `Country` from internal representation lies with the api consumer, rather than trying to preempt a dc-specific implementation